src/IssueViz.UnitTests/Hotspots/HotspotsToolWindowCommandTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Design;
+using FluentAssertions;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.IssueVisualization.Hotspots.Commands;
+using SonarLint.VisualStudio.IssueVisualization.Hotspots.HotspotsControl;
+using SonarLint.VisualStudio.IssueVisualization.UnitTests.Helpers;
+
+namespace SonarLint.VisualStudio.IssueVisualization.UnitTests.Hotspots
+{
+    [TestClass]
+    public class HotspotsToolWindowCommandTests : ToolWindowCommandTests<HotspotsToolWindow>
+    {
+        public HotspotsToolWindowCommandTests() 
+            : base(ExecuteCommand)
+        {
+        }
+
+        private static void ExecuteCommand(AsyncPackage package, ILogger logger)
+        {
+            var testSubject = new HotspotsToolWindowCommand(package, Mock.Of<IMenuCommandService>(), logger);
+            
+            testSubject.Execute(null, EventArgs.Empty);
+        }
+
+        [TestMethod]
+        public void Ctor_ArgsCheck()
+        {
+            var package = Mock.Of<AsyncPackage>();
+            var commandService = Mock.Of<IMenuCommandService>();
+            var logger = Mock.Of<ILogger>();
+
+            Action act = () => new HotspotsToolWindowCommand(null, commandService, logger);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("package");
+
+            act = () => new HotspotsToolWindowCommand(package, null, logger);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("commandService");
+
+            act = () => new HotspotsToolWindowCommand(package, commandService, null);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("logger");
+        }
+
+        [TestMethod]
+        public void Ctor_CommandAddedToMenu()
```

Comment:
I'd have thought that this would have been a common test too.

Reply:
IssueViz adds multiple commands so I left them separated. I could pass to base CommandSet + list of commands ids, WDYT?

Reply:
That's the sort of thing I had in mind, or even just having a common assertion method like `CheckCommandRegistered(guid, id)`, rather than repeating the `commandService.Verify` code in each test.

Reply:
Ok, this PR is merged so I'll do that in separate PR